### PR TITLE
Adds getByName to RecordClient

### PR DESCRIPTION
**Pull request by ecloud-jay**
Added as part of task to add RDNS to FIPS.

### CODE REVIEW

File: src/SafeDNS/RecordClient.php [hunk 1]
```diff
@@ -55,6 +55,22 @@ public function getZoneRecordById($zoneName, $id)
...
+     * Gets a specific zone record by name
+     *
+     * @param $zoneName
+     * @param string $id
+     * @return Record
+     * @throws \GuzzleHttp\Exception\GuzzleException
+     */
+    public function getByName($name)
+    {
+        $response = $this->request("GET", sprintf("v1/records?name=%s", $name));
```
Comment (ptmcnally) [inline src/SafeDNS/RecordClient.php:68]: This is not valid APIO filtering, see https://developers.ukfast.io/api-guide

Comment (ptmcnally) [reply src/SafeDNS/RecordClient.php:68]: ...also, would this not return a collection instead of a single resource?

**Review by ptmcnally — COMMENTED** [review]
Does not implement valid API filtering.

File: src/SafeDNS/RecordClient.php (hunk 1 above)
Comment (ptmcnally) [reply src/SafeDNS/RecordClient.php:68]: As this is an index/collection endpoint you may be better implementing a getPage() method like this https://github.com/ukfast/sdk-php/blob/master/src/SafeDNS/ZoneClient.php#L24, then pass in your name filter wherever you are using it.

Comment (ecloud-jay) [reply src/SafeDNS/RecordClient.php:68]: v1/records only returns one record when the exact name is in, similar to a getById, would I not just have to change line 71 to $body->data[0] if it exists?

**Comment by ecloud-jay** [comment]
Not needed

**Pull request closed without merging**